samples/sensor/bmi270/src/main.c

```diff
@@ -11,19 +11,16 @@
 
 void main(void)
 {
-	const struct device *dev;
+	const struct device *dev = DEVICE_DT_GET_ANY(bosch_bmi270);
 	struct sensor_value acc[3], gyr[3];
 	struct sensor_value full_scale, sampling_freq, oversampling;
 
-	dev = device_get_binding(DT_LABEL(DT_INST(0, bosch_bmi270)));
-	if (dev == NULL) {
-		printf("Could not get %s device\n",
-		       DT_LABEL(DT_INST(0, bosch_bmi270)));
+	if (!device_is_ready(dev)) {
+		printf("Device %s is not ready\n", dev->name);
```

Comment:
note: because you're using `DEVICE_DT_GET_ANY` you could reach this point because there's no device with `bosch_bmi270` (dev == NULL), not only because it's not ready. I think that to be consistent with DEVICE_DT_GET/DEVICE_DT_GET_OR_NULL we should probably rename DEVICE_DT_GET_ANY to DEVICE_DT_GET_ONE_OR_NULL. The ONE/ANY naming is confusing.

Reply:
`DEVICE_DT_GET_ANY` will return `NULL` if there's no device with the given compatible. If I remember correctly, it was added because it makes samples _friendlier_ by failing at runtime instead. However, we loose CI coverage by using it... And as said, its name is confusing IMO.

Reply:
I agree, I expected `DEVICE_DT_GET_ANY` to fail at compile-time without the given compatible.